### PR TITLE
Add the typeclass IsSendable

### PR DESCRIPTION
**Pull request by gbagan**
This refers to the issue: https://github.com/purescript-web/purescript-web-workers/issues/2

Add the typeclass IsSendable whose instances are Boolean, Int, Number, Char, String, Array, Record.
Change the signature of postMessage with
```haskell
postMessage :: forall msg. IsSendable msg => msg -> Worker -> Effect Unit
``` 

**Prerequisites**

- [X] Before opening a pull request, please check the DOM standard (https://dom.spec.whatwg.org/). If it doesn't appear in this spec, it may be present in the spec for one of the other `purescript-web` projects. Although MDN is a great resource, it is not a suitable reference for this project.

**Description of the change**

Clearly and concisely describe the purpose of the pull request. If this PR relates to an existing issue or change proposal, please link to it. Include any other background context that would help reviewers understand the motivation for this PR.

---

**Checklist:**

- [X] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [X] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Notable changes to this project are documented in this file. The format is based
 ## [Unreleased]
 
 Breaking changes:
+- The data argument of functions `postMessage` and `postMessage'` must be instance of a new typeclass `isSendable`
```
Comment (JordanMartinez) [inline CHANGELOG.md:8]: Typo on `isSendable` -> `IsSendable`

File: src/Web/Worker/Types.purs [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+instance IsSendable Number
+instance IsSendable Char
+instance IsSendable String
+instance IsSendable a => IsSendable (Array a)
+instance (RowToList r rl, IsSendableRowList rl) => IsSendable (Record r)
+
+class IsSendableRowList (rl :: RowList Type)
+
+instance IsSendableRowList Row.Nil
+instance (IsSendable a, IsSendableRowList rest) => IsSendableRowList (Row.Cons sym a rest)
```
Comment (JordanMartinez) [inline src/Web/Worker/Types.purs:28]: You didn't port over the `Sendable` newtype that Nate had?

**Review by JordanMartinez — COMMENTED** [review]
LGTM aside from the below things.

**Comment by gbagan** [comment]
@JordanMartinez 
I have applied your suggestions.

File: src/Web/Worker/Types.purs [hunk 3]
```diff
@@ -21,8 +24,24 @@ instance IsSendable Char
 instance IsSendable String
 instance IsSendable a => IsSendable (Array a)
 instance (RowToList r rl, IsSendableRowList rl) => IsSendable (Record r)
+instance IsSendable Unit
+instance IsSendable Void
```
Comment (JordanMartinez) [inline src/Web/Worker/Types.purs:32]: Can you clarify why you added an instance for `Void`? I'm not sure that's correct, but I'm not sure about that either.

File: src/Web/Worker/Types.purs [hunk 4]
```diff
@@ -21,8 +24,24 @@ instance IsSendable Char
 instance IsSendable String
 instance IsSendable a => IsSendable (Array a)
 instance (RowToList r rl, IsSendableRowList rl) => IsSendable (Record r)
+instance IsSendable Unit
```
Comment (JordanMartinez) [inline src/Web/Worker/Types.purs:31]: Why is there an instance for `Unit`? Can you clarify?

File: src/Web/Worker/Types.purs [hunk 5]
```diff
@@ -21,8 +24,24 @@ instance IsSendable Char
...
+wrap :: forall a b. Newtype a b => IsSendable b => a -> SendWrapper a
+wrap = SendWrapper
+
+unwrap :: forall a. SendWrapper a -> a
+unwrap (SendWrapper a) = a
+
+-- | Use with care. If you send something that isn't actually Sendable, it
+-- | will raise an exception.
+unsafeWrap :: forall a. a -> SendWrapper a
+unsafeWrap = SendWrapper
```
Comment (JordanMartinez) [inline src/Web/Worker/Types.purs]: Minor nit, but could a newline be added here? It helps with removing diffs if someone should add content below `unsafeWrap`

Comment (JordanMartinez) [reply src/Web/Worker/Types.purs]: Ah, nevermind. A later commit had this.

File: src/Web/Worker/Types.purs (hunk 4 above)
Comment (gbagan) [reply src/Web/Worker/Types.purs:31]: Actually, this instance and the instance for Void comes from the library workerbees but I can remove them if you want.

Comment (JordanMartinez) [reply src/Web/Worker/Types.purs:31]: Ah, ok. I think I missed that.

**Pull request merged**